libraries/rush-sdk/src/typings.d.ts

```diff
@@ -0,0 +1 @@
+declare module '@microsoft/rush-lib/lib-esnext/api/RushGlobalFolder';
```

Comment:
Why is this necessary?

Reply:
If I removed this declaration, tsc for `rush-sdk` complains the following error. I suppose types are not generated under `rush-lib`'s `lib-esnext` folder.

![CleanShot 2024-06-24 at 11 00 55@2x](https://github.com/microsoft/rushstack/assets/16147702/5584e11b-ed87-4dbf-8703-a43905ae7b1e)

Reply:
This can be solved by fiddling with `typesVersions` in package.json in rush-lib:
```json
"typesVersions": {
    "*": {
      "lib-esnext/*": [
        "lib/*"
      ]
    }
  }
```
Or similar mapping.

Reply:
Thanks @dmichon-msft "typesVersions" works like a charm!

I tested it locally and `tsc` can handle `rush-lib`'s deep imports well after adding `typesVerions` in package.json of `rush-lib`

Code updated to reflect this change.

cc: @iclanton